File: boilerplate/backend/shared/go/recipes/all_auth.go
package main

import (
	"fmt"
	"os"

	"github.com/supertokens/supertokens-golang/recipe/dashboard"
	"github.com/supertokens/supertokens-golang/recipe/emailpassword"
	"github.com/supertokens/supertokens-golang/recipe/passwordless"
	"github.com/supertokens/supertokens-golang/recipe/passwordless/plessmodels"
	"github.com/supertokens/supertokens-golang/recipe/session"
	"github.com/supertokens/supertokens-golang/recipe/thirdparty"
	"github.com/supertokens/supertokens-golang/recipe/thirdparty/tpmodels"
	"github.com/supertokens/supertokens-golang/recipe/userroles"
	"github.com/supertokens/supertokens-golang/supertokens"
)

func getStringPointer(s string) *string {
    return &s
}

func getApiDomain() string {
    apiPort := os.Getenv("VITE_APP_API_PORT")
    if apiPort == "" {
        apiPort = "3001"
    }
    apiUrl := os.Getenv("VITE_APP_API_URL")
    if apiUrl == "" {
        apiUrl = fmt.Sprintf("http://localhost:%s", apiPort)
    }
    return apiUrl
}

func getWebsiteDomain() string {
    websitePort := os.Getenv("VITE_APP_WEBSITE_PORT")
    if websitePort == "" {
        websitePort = "3000"
    }
    websiteUrl := os.Getenv("VITE_APP_WEBSITE_URL")
    if websiteUrl == "" {
        websiteUrl = fmt.Sprintf("http://localhost:%s", websitePort)
    }
    return websiteUrl
}

// SuperTokensConfig is the configuration for SuperTokens core with all auth methods
var SuperTokensConfig = supertokens.TypeInput{
    Supertokens: &supertokens.ConnectionInfo{
        ConnectionURI: "https://try.supertokens.com",
    },
    AppInfo: supertokens.AppInfo{
        AppName:         "SuperTokens Demo App",
        APIDomain:       getApiDomain(),
        WebsiteDomain:   getWebsiteDomain(),
        APIBasePath:     getStringPointer("/auth"),
        WebsiteBasePath: getStringPointer("/auth"),
    },
    RecipeList: []supertokens.Recipe{
        emailpassword.Init(nil),
        passwordless.Init(plessmodels.TypeInput{
            FlowType: "USER_INPUT_CODE_AND_MAGIC_LINK",
            ContactMethodEmailOrPhone: plessmodels.ContactMethodEmailOrPhoneConfig{
                Enabled: true,
            },
        }),
        thirdparty.Init(&tpmodels.TypeInput{
            SignInAndUpFeature: tpmodels.TypeInputSignInAndUp{
                Providers: []tpmodels.ProviderInput{
                    {
                        Config: tpmodels.ProviderConfig{
                            ThirdPartyId: "google",
                            Clients: []tpmodels.ProviderClientConfig{
                                {
                                    ClientID:     "1060725074195-kmeum4crr01uirfl2op9kd5acmi9jutn.apps.googleusercontent.com",
                                    ClientSecret: "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",
                                },
                            },
                        },
                    },
                    {
                        Config: tpmodels.ProviderConfig{
                            ThirdPartyId: "github",
                            Clients: []tpmodels.ProviderClientConfig{
                                {
                                    ClientID:     "467101b197249757c71f",
                                    ClientSecret: "e97051221f4b6426e8fe8d51486396703012f5bd",
                                },
                            },
                        },
                    },
                    {
                        Config: tpmodels.ProviderConfig{
                            ThirdPartyId: "apple",
                            Clients: []tpmodels.ProviderClientConfig{
                                {
                                    ClientID: "4398792-io.supertokens.example.service",
                                    AdditionalConfig: map[string]interface{}{
                                        "keyId":      "7M48Y4RYDL",
                                        "privateKey": "-----BEGIN PRIVATE KEY-----\nMIGTAgEAMBMGByqGSM49AgEGCCqGSM49AwEHBHkwdwIBAQQgu8gXs+XYkqXD6Ala9Sf/iJXzhbwcoG5dMh1OonpdJUmgCgYIKoZIzj0DAQehRANCAASfrvlFbFCYqn3I2zeknYXLwtH30JuOKestDbSfZYxZNMqhF/OzdZFTV0zc5u5s3eN+oCWbnvl0hM+9IW0UlkdA\n-----END PRIVATE KEY-----",
                                        "teamId":     "YWQCXGJRJL",
                                    },
                                },
                            },
                        },
                    },
                    {
                        Config: tpmodels.ProviderConfig{
                            ThirdPartyId: "twitter",
                            Clients: []tpmodels.ProviderClientConfig{
                                {
                                    ClientID:     "4398792-WXpqVXRiazdRMGNJdEZIa3RVQXc6MTpjaQ",
                                    ClientSecret: "BivMbtwmcygbRLNQ0zk45yxvW246tnYnTFFq-LH39NwZMxFpdC",
                                },
                            },
                        },
                    },
                },
            },
        }),
        session.Init(nil),
        dashboard.Init(nil),
        userroles.Init(nil),
    },
} 

File: boilerplate/backend/shared/python/recipes/all_auth.py
from supertokens_python import init, InputAppInfo, SupertokensConfig
from supertokens_python.recipe import (
    thirdpartyemailpassword,
    thirdpartypasswordless,
    session,
    dashboard,
    userroles
)
from supertokens_python.recipe.thirdparty.provider import GoogleProvider, GithubProvider, AppleProvider
from supertokens_python.recipe.passwordless import ContactEmailOrPhoneConfig
import os

def get_api_domain() -> str:
    api_port = os.environ.get("VITE_APP_API_PORT", "3001")
    api_url = os.environ.get("VITE_APP_API_URL", f"http://localhost:{api_port}")
    return api_url

def get_website_domain() -> str:
    website_port = os.environ.get("VITE_APP_WEBSITE_PORT", "3000")
    website_url = os.environ.get("VITE_APP_WEBSITE_URL", f"http://localhost:{website_port}")
    return website_url

# Configuration for SuperTokens core and application
supertokens_config = SupertokensConfig(
    connection_uri="https://try.supertokens.com"
)

app_info = InputAppInfo(
    app_name="SuperTokens Demo App",
    api_domain=get_api_domain(),
    website_domain=get_website_domain()
)

framework = None

recipe_list = [
    thirdpartyemailpassword.init(
        providers=[
            GoogleProvider(
                client_id=os.environ.get("GOOGLE_CLIENT_ID", "1060725074195-kmeum4crr01uirfl2op9kd5acmi9jutn.apps.googleusercontent.com"),
                client_secret=os.environ.get("GOOGLE_CLIENT_SECRET", "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")
            ),
            GithubProvider(
                client_id=os.environ.get("GITHUB_CLIENT_ID", "467101b197249757c71f"),
                client_secret=os.environ.get("GITHUB_CLIENT_SECRET", "e97051221f4b6426e8fe8d51486396703012f5bd")
            ),
            AppleProvider(
                client_id=os.environ.get("APPLE_CLIENT_ID", "4398792-io.supertokens.example.service"),
                key_id=os.environ.get("APPLE_KEY_ID", "7M48Y4RYDL"),
                private_key=os.environ.get("APPLE_PRIVATE_KEY", "-----BEGIN PRIVATE KEY-----\nMIGTAgEAMBMGByqGSM49AgEGCCqGSM49AwEHBHkwdwIBAQQgu8gXs+XYkqXD6Ala9Sf/iJXzhbwcoG5dMh1OonpdJUmgCgYIKoZIzj0DAQehRANCAASfrvlFbFCYqn3I2zeknYXLwtH30JuOKestDbSfZYxZNMqhF/OzdZFTV0zc5u5s3eN+oCWbnvl0hM+9IW0UlkdA\n-----END PRIVATE KEY-----").replace("\\n", "\n"),
                team_id=os.environ.get("APPLE_TEAM_ID", "YWQCXGJRJL")
            )
        ]
    ),
    thirdpartypasswordless.init(
        flow_type="USER_INPUT_CODE_AND_MAGIC_LINK",
        contact_config=ContactEmailOrPhoneConfig(),
        providers=[
            GoogleProvider(
                client_id=os.environ.get("GOOGLE_CLIENT_ID", "1060725074195-kmeum4crr01uirfl2op9kd5acmi9jutn.apps.googleusercontent.com"),
                client_secret=os.environ.get("GOOGLE_CLIENT_SECRET", "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")
            ),
            GithubProvider(
                client_id=os.environ.get("GITHUB_CLIENT_ID", "467101b197249757c71f"),
                client_secret=os.environ.get("GITHUB_CLIENT_SECRET", "e97051221f4b6426e8fe8d51486396703012f5bd")
            ),
            AppleProvider(
                client_id=os.environ.get("APPLE_CLIENT_ID", "4398792-io.supertokens.example.service"),
                key_id=os.environ.get("APPLE_KEY_ID", "7M48Y4RYDL"),
                private_key=os.environ.get("APPLE_PRIVATE_KEY", "-----BEGIN PRIVATE KEY-----\nMIGTAgEAMBMGByqGSM49AgEGCCqGSM49AwEHBHkwdwIBAQQgu8gXs+XYkqXD6Ala9Sf/iJXzhbwcoG5dMh1OonpdJUmgCgYIKoZIzj0DAQehRANCAASfrvlFbFCYqn3I2zeknYXLwtH30JuOKestDbSfZYxZNMqhF/OzdZFTV0zc5u5s3eN+oCWbnvl0hM+9IW0UlkdA\n-----END PRIVATE KEY-----").replace("\\n", "\n"),
                team_id=os.environ.get("APPLE_TEAM_ID", "YWQCXGJRJL")
            )
        ]
    ),
    session.init(),
    dashboard.init(),
    userroles.init()
] 